### DISCUSSION TITLE
Error when compiling

### DISCUSSION THREAD

**Discussion by lilei199908** [idaholab/moose#26584] [Q&A Getting Started] [answered]
When I followed the example
`cd ~/projects/YourAppName
make -j 6
./run_tests -j 6`
My makefile file is
`ALL_MODULES                 := no

CHEMICAL_REACTIONS          := no
CONTACT                     := no
ELECTROMAGNETICS            := no
EXTERNAL_PETSC_SOLVER       := no
FLUID_PROPERTIES            := no
FSI                         := no
FUNCTIONAL_EXPANSION_TOOLS  := no
GEOCHEMISTRY                := no
HEAT_TRANSFER               := yes
LEVEL_SET                   := no
MISC                        := no
NAVIER_STOKES               := no
OPTIMIZATION                := no
PERIDYNAMICS                := no
PHASE_FIELD                 := yes
POROUS_FLOW                 := no
RAY_TRACING                 := no
REACTOR                     := no
RDG                         := no
RICHARDS                    := no
STOCHASTIC_TOOLS            := no
THERMAL_HYDRAULICS          := no
TENSOR_MECHANICS            := no
XFEM                        := no
`
report an error
`Linking Executable /home/lilei2023/projects/your_app_name/heat_transfer-opt...
/home/lilei2023/miniforge/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/lilei2023/projects/moose/modules/module_loader/lib/libmodule_loader_with_ray_ht_tm_pf-opt.so: undefined reference to `HeatTransferApp::registerExecFlags(Factory&)'
/home/lilei2023/miniforge/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/lilei2023/projects/your_app_name/lib/libheat_transfer-opt.so: undefined reference to `HeatTransferApp::registerAll(Factory&, ActionFactory&, Syntax&)'
/home/lilei2023/miniforge/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/lilei2023/projects/moose/modules/module_loader/lib/libmodule_loader_with_ray_ht_tm_pf-opt.so: undefined reference to `HeatTransferApp::associateSyntax(Syntax&, ActionFactory&)'
/home/lilei2023/miniforge/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/lilei2023/projects/moose/modules/module_loader/lib/libmodule_loader_with_ray_ht_tm_pf-opt.so: undefined reference to `HeatTransferApp::registerObjects(Factory&)'
/home/lilei2023/miniforge/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/lilei2023/projects/moose/modules/module_loader/lib/libmodule_loader_with_ray_ht_tm_pf-opt.so: undefined reference to `HeatTransferApp::registerApps()'
collect2: error: ld returned 1 exit status
make: *** [/home/lilei2023/projects/moose/framework/app.mk:445: /home/lilei2023/projects/your_app_name/heat_transfer-opt] Error 1
`

**Comment by GiudGiud** [comment, marked as answer]
Hello

We just changed a few things about the main app source files. 

You will need to save all your work using git commit then clean the entire repo with git clean -xfd run at the moose/ folder level

Then recompile

Guillaume

**Reply by lilei199908** [reply]
I re-cloned it and it compiled successfully, thanks!
